src/java.base/share/classes/java/lang/Class.java

```diff
@@ -801,8 +803,9 @@ public T newInstance()
      *          {@code false} otherwise.
      * @since   1.1
      */
-    @IntrinsicCandidate
-    public native boolean isArray();
+    public boolean isArray() {
+        return componentType != null;
```

Comment:
The componentType declaration should have a comment indicating that == null is the sole indication that the class is an interface.
Perhaps there should be an assert somewhere validating/cross checking that requirement.

Reply:
I added an assert for set_component_mirror() in the vm, but I don't see how to assert it in Java.  Is the comment like:

// The componentType field's null value is the sole indication that the class is an array, see isArray()